tests/testthat/test-linear_pool.R

```diff
@@ -411,6 +418,129 @@ test_that("(weighted) quantiles correctly calculated - lognormal family", {
 })
 
 
+test_that("Not all component models forecasting for the same set of dependent task values throws an error", {
+  sample_outputs <- expand.grid(stringsAsFactors = FALSE,
```

Comment:
It may be too late now for this to be a useful suggestion, but since we use this example repeatedly, it could be helpful to pull its definition out into a helper function.

Reply:
I do think this would be useful to do: it would make this file shorter, and it would make it easier to find the difference in the setup for each test case.  For example, in this test we would take a complete, valid submission with both horizons and then "break" it by filtering it on the combination of `model_id` and `horizon`.

One way to achieve this could be to create a file called `tests/testthat/helper.R` or `tests/testthat/helper-test_data.R`, or something else like that, and within that define a function like `create_test_sample_outputs()` that creates a generally useful/usable example of `sample_outputs`.  Then in this function, you can call `sample_outputs <- create_test_sample_outputs()` and modify it as needed.  For example, you might have something like:
```r
sample_outputs <- create_test_sample_outputs() |>
  dplyr::filter(model_id %in% letters[1:3] | horizon == 1)
```

This setup is described in the R packages book in these two sections:
 - [helper files](https://r-pkgs.org/testing-design.html#testthat-helper-files)
 - [test fixtures](https://r-pkgs.org/testing-advanced.html#test-fixtures)